crates/bevy_ecs/src/query/state.rs

```diff
@@ -337,6 +343,12 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
         }
     }
 
+    /// Gets the query result for the given [`World`] and [`Entity`].
+    /// This will only work for archetypes that this [`QueryState`] has in its cache.
+    /// To ensure that the cache is up to date, call [`QueryState::update_archetypes`].
+    /// Alternatively, call [`QueryState::get`] instead, which calls `update_archetypes` for you.
```

Comment:
I'd like to see something along the line of "The `QueryState` cache is populated in `QueryState::new`, if `world` changed since then, you should call `update_archetypes`."

I earlier had to look at the source code of `QueryState::new` because this isn't explicit, and the docs kinda implies you _always_ need to call `update_archetypes` after  `QueryState::new`, which is not the case.